SofaKernel/extlibs/newmat/CMakeLists.txt

```diff
@@ -42,6 +42,9 @@ add_library(${PROJECT_NAME} STATIC ${HEADER_FILES} ${SOURCE_FILES})
 target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
 target_include_directories(${PROJECT_NAME} PUBLIC "$<INSTALL_INTERFACE:include>")
 target_compile_definitions(${PROJECT_NAME} PRIVATE "use_namespace")
+if(SOFA_FLOAT)
+    target_compile_definitions(${PROJECT_NAME} PRIVATE "SOFA_FLOAT")
+endif()
```

Comment:
I initially tried the method advocated on the newmat library FAQ [see 2.3 Customizing](http://robertnz.net/nm11.htm). It was not working, because the newmat/include.h file was modified by sofa developers at some point, who also added the sofa macro inside, see line 31. In the end, SOFA_FLOAT in include.h is only a proxy to other define USING_DOUBLE and USING_FLOAT, so it could probably be removed.